src/Uno.UI/UI/Xaml/Controls/CalendarView/CalendarViewBaseItemChrome.uno.cs

```diff
@@ -56,8 +58,9 @@ internal override void OnArrangeVisual(Rect rect, Rect? clip)
 
 		private void UpdateChromeIfNeeded(Rect rect)
 		{
-			if (rect.Width > 0 && rect.Height > 0 && _lastSize != rect.Size)
+			if (rect.Width > 0 && rect.Height > 0 && (_lastSize != rect.Size || _forceUpdateChrome))
```

Comment:
This code is confusing me a bit. If `_forceUpdateChrome` is true, then we know `_lastSize` is `default`, and it will remain `default` until `UpdateChromeIfNeeded` is called.
So, now, if we get to `UpdateChromeIfNeeded` with true `_forceUpdateChrome`, then we already know `_lastSize` is default, in which case `_lastSize != rect.Size` is already true because `default != some_size_with_positive_width_and_height`

Put in other words, if `Uno_InvalidateRender` is called, then the next `UpdateChromeIfNeeded` is guaranteed to do the update, without the need for an additional flag.

This is at least my understanding of the code, please correct me if I'm missing something.

Reply:
`InvalidateRender` is called even in case the `_lastSize` is not `default` - and actually hasn't changed from last time - hence the flag is needed to force the render for that specific case.

So in some cases the invalidation is needed because the previous size was default, and in others because the theme changed for example and we need to invalidate the background, border, etc.

Reply:
@MartinZikmund But the only code path that sets `_forceUpdateChrome` to true also sets `_lastSize` to default. So if we set `_forceUpdateChrome` to true, we know `_lastSize` was set to default (lines 21 and 22 in this file).

---

> InvalidateRender is called even in case the _lastSize is not default - and actually hasn't changed from last time - hence the flag is needed to force the render for that specific case.

If `_lastSize` is not default, then `_forceUpdateChrome` will be false, per my understanding of the code.

`_lastSize` and `_forceUpdateChrome` are always set together in lines 21-22 and 63-64.

Reply:
Indeed! I have added the condition because of another change, which I undone later and that meant the condition was superfluous. 👍